file-upload-example/README.md

```diff
@@ -0,0 +1,139 @@
+
+Here is an example that I used:
+
+```ts
+export const environment = {
+  production: false,
+  endpoint: 'http://localhost/v1', // replace with your own appwrite server ip
+  project: '5f7860217c7e7' // replace with your own appwrite project id
+};
+```
+
+## Working with Appwrite
+
+For working with Appwrite we will use a a service (`src/app/appwrite.service.ts`). Most of the service's methods is just a wrapper around Appwrite's SDK Storage API methods.   
+I'll wrap every promise around an `Observable` to be consistent with Angular's built-in `HttpClient`.
+
+### Detail about our service
+
+#### Initializing the SDK
+
+```ts
+  // ...
+  constructor() {
+    this.sdk.setEndpoint(environment.endpoint);
+    this.sdk.setProject(environment.project);
+  }
+  // ...
+```
+
+With this as soon as the service is creating the `endpoint` and `project` is also set and since our service is provided in root it will be only created once across the whole app. 
+
+Removing and uploading a file requires a valid user session, we are only focusing on Storage API in this tutorial so we will use an already created user through the Appwrite console. __Please make sure__ to replace `email` and `password` in `dummyLogin` function.
+
+#### Uploading a file
+
+```ts
+  public uploadFile(file: File): Observable<object> {
+    return from(this.sdk.storage.createFile(file, ['*'], ['*']));
+  }
+```
+
+We just need to pass through our `File` to upload and a set permissions. The first array is an array for `read` permissions, the second is for `write`. For this tutorial we wil use `*` (wildcard) which means this is available publicly. You can learn more about permissions [here](https://appwrite.io/docs/permissions).
+
+#### Listing the files
+
+```ts
+  public listAllStorageFiles(): Observable<object> {
+    return from(this.sdk.storage.listFiles(null, null, null, null));
+  }
+```
+
+To list files you can pass more parameter for filtering, paging and ordering, but for this tutorial we will list everything.
+
+#### Removing a file
+
+```ts
+  public removeFile(id: string): Observable<object> {
+    return from(this.sdk.storage.deleteFile(id));
+  }
+```
+
+To remove a file you only need to pass the file's `id`.
+
+#### Previewing a file
+
+```ts
+  public previewFile(id: string): string {
+    return this.sdk.storage.getFilePreview(id, 100, 100, null, 'fff', null);
+  }
+```
+
+To preview a file you'll need the `fileId` and optionally you can also set `width`, `height`, `quality`, `backgroundColor` (hex code without `#`) and `output` to convert your file into your needs. For now we will request a `100×100` preview on a white background.
+
+#### Downloading a file
+
+```ts
+  public downloadFileLink(id: string): string {
+    return this.sdk.storage.getFileDownload(id);
+  }
+```
+
+To get a download link we'll need the `fileId` and this will return a simple string that we can bind to an `a` tag `href` property.
```

Comment:
```suggestion
To get a download link, we'll need the `fileId` and this will return a simple string that we can bind to an `a` tag `href` property.
```